provider/resource_harbor_project.go

```diff
@@ -114,8 +123,29 @@ func resourceProjectUpdate(d *schema.ResourceData, meta interface{}) error {
 
 func resourceProjectDelete(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*harbor.Client)
+	projectName := d.Get("name").(string)
+
+	repos, err := client.GetRepositories(d.Get("projectid").(string))
+	if err != nil {
+		return err
+	}
+
+	err = client.DeleteRepositories(repos)
+	if err != nil {
+		return err
+	}
+
+	charts, err := client.GetCharts(projectName)
+	if err != nil {
+		return err
+	}
+
+	err = client.DeleteCharts(projectName, charts)
```

Comment:
same as above, let's check if `charts` is empty before making this call

Reply:
Done